hyperapp.d.ts

```diff
@@ -0,0 +1,45 @@
+export as namespace hyperapp
+
+// VNode part
+
+export interface VNode<Data> {
+  tag: string
+  data?: Data
+  children: VElement<{}>[]
+}
+
+export type VElement<Data> = VNode<Data> | string
```

Comment:
You want a VNode to be either a VNode or a string. I never heard it was possible. Union is only possible on type.

I guess, it is right to use a type.
A VirtualNode is a `VirtualNode` and a `VirtualElement` could be either a `VirtualNode` or a `string`.